### PR TITLE
Create jinja_partials recipe

### PR DESCRIPTION
**Pull request by Colelyman**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Colelyman** [comment]
I'm willing to be a maintainer!

**Comment by jakirkham** [comment]
The missing `requirements.txt` is caused by an upstream packaging issue: https://github.com/mikeckennedy/jinja_partials/issues/9

File: recipes/jinja_partials/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{% set name = "jinja_partials" %}
+{% set version = "0.2.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/jinja_partials-{{ version }}.tar.gz
```
Comment (bollwyvl) [inline recipes/jinja_partials/meta.yaml:9]: The `.tar.gz` doesn't include `requirements.txt`. 

Ideally, you'd request this be fixed upstream (probably a single-line change `MANIFEST.in`), but in the meantime:

```yaml
source:
  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/jinja_partials-{{ version }}.tar.gz
    sha256: ada44c441fe175d0a5d5f594b5f4c3301a1d5974d212e83031fbd8886785dd69
  - url: https://raw.githubusercontent.com/mikeckennedy/jinja_partials/v{{ version }}/requirements.txt
    sha256: 3e47955ad2dbdfe0aa8d68b57cedbd3826e3e791215ff72184aac276b55eff3b
```

Comment (Colelyman) [reply recipes/jinja_partials/meta.yaml]: Thanks @bollwyvl! I mistakenly checked the `.tar.gz` from the Github tag and saw that `requirements.txt` is present, so I ended up changing the original source to that.

**Comment by Colelyman** [comment]
@conda-forge/help-python ready for review! Thank you

**Comment by bollwyvl** [comment]
Sure, whatever floats your boat. Making the upstream PR is still recommended, as, long term, PyPI releases are a more reliable distribution: GitHub archives can change their `sha256` at any time.

**Comment by Colelyman** [comment]
@bollwyvl Thanks, I didn't know that the GitHub archives can change their `sha256`. We will see what the maintainer says in my upstream PR https://github.com/mikeckennedy/jinja_partials/pull/8

**Comment by bollwyvl** [comment]
That PR seems fairly complex and partially defeats the purpose of having
data-driven packaging.

The simpler play for old style packaging is a creating MANIFEST.in that
lists `include requirements.txt` (and probably LICENSE)

The better play, generally, is moving to pyproject.toml, even if the
build-system is still setuptools.

**Comment by Colelyman** [comment]
There has been a new release, and my hacky PR wasn't merged in (thankfully). So I think now everything is good to go!

File: recipes/jinja_partials/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  sha256: 5b32cc035ee93f372ddfcecf96c4837450d8561d0e2ff26c89dc0d0bd071128b
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  host:
+    - python >=3.6
```
Comment (jakirkham) [inline recipes/jinja_partials/meta.yaml:19]: Looks like they will [bump to Python 3.8 in the next release]( https://github.com/mikeckennedy/jinja_partials/commit/285d147ad3867c4884de414fea36b928e68364d0#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R6 ). So please keep an eye on that

**Review by jakirkham — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Cole and Nick! 🙏

**Comment by Colelyman** [comment]
Thanks for your help @jakirkham and @bollwyvl!

**Pull request merged**